bigbang/archive.py

```diff
@@ -340,3 +357,179 @@ def clean_footer(foot):
+    indexed by message-id, with column-names from headers.
+    """
+    # extract data into a list of tuples -- records -- with
+    # the Message-ID separated out as an index
+    # valid_messages = [m for m in messages if m.get()
+
+    pm = [
+        (
+            m.get("Message-ID"),
+            str(m.get("From")).replace("\\", " "),
+            str(m.get("Subject")),
+            get_date(m),
+            str(m.get("In-Reply-To")),
+            str(m.get("References")),
+            get_text(m),
+        )
+        for m in messages
+        if m.get("From")
+    ]
+
+    mdf = pd.DataFrame.from_records(
+        list(pm),
+        index="Message-ID",
+        columns=[
+            "Message-ID",
+            "From",
+            "Subject",
+            "Date",
+            "In-Reply-To",
+            "References",
+            "Body",
+        ],
+    )
+    mdf.index.name = "Message-ID"
+
+    return mdf
+
+
+def open_list_archives(
+    url: str,
+    archive_dir: str = CONFIG.mail_path,
+    mbox: bool = False,
+) -> pd.DataFrame:
+    """
+    Return a list of all email messages contained in the specified directory.
+
+    Parameters
+    -----------
+    url: str
+        the name of a subdirectory of the directory specified
+        in argument *archive_dir*. This directory is expected to contain
+        files with extensions .txt, .mail, or .mbox. These files are all
+        expected to be in mbox format-- i.e. a series of blocks of text
+        starting with headers (colon-separated key-value pairs) followed by
+        an email body.
+        TODO: this argument should be re-named. sometimes it's being called with
+        actual URLs, other times with subdirectory names, leading to spurious
+
+    archive_dir : str:
+        directory containing all messages.
+
+    mbox: bool, default False
+        True if there's an mbox file already available for this archive.
+
+    Returns
+    --------
+
+    data : pandas.DataFrame
+
+    """
+    if (url is None) and (archive_dir is None):
+        raise ValueError("Either `url` or `archive_dir` must be given.")
+
+    messages = None
+
+    if mbox and (os.path.isfile(os.path.join(archive_dir, url))):
+        # treat string as the path to a file that is an mbox
+        box = mailbox.mbox(os.path.join(archive_dir, url), create=False)
+        messages = list(box.values())
+
+    else:
+        # assume string is the path to a directory with many
+        # TODO: Generalize get_list_name to listserv, w3c?
+        list_name = mailman.get_list_name(url)
+        arc_dir = archive_directory(archive_dir, list_name)
+
+        file_extensions = [".txt", ".mail", ".mbox"]
+
+        txts = [
+            os.path.join(arc_dir, fn)
+            for fn in os.listdir(arc_dir)
+            if any([fn.endswith(extension) for extension in file_extensions])
+        ]
+
+        print("^^^^^^^^^^^^^")
+        print(os.listdir(arc_dir))
```

Comment:
I think we shouldn't be using `print` statements for this kind of status/debugging info. Can the `logging.info` statement be updated to include whatever is needed here?